packages/backend-core/src/sql/utils.ts

```diff
@@ -8,7 +8,7 @@ const ROW_ID_REGEX = /^\[.*]$/g
 const ENCODED_SPACE = encodeURIComponent(" ")
 
 export function isExternalTableID(tableId: string) {
-  return tableId.includes(DocumentType.DATASOURCE)
+  return tableId.startsWith(DocumentType.DATASOURCE + SEPARATOR)
```

Comment:
Is this _definitely_ correct? Seems it would be a pretty big problem if not.

Reply:
Did this in response to: https://github.com/Budibase/budibase/pull/13735#discussion_r1611382875 - table IDs come from here: https://github.com/Budibase/budibase/blob/11f3d9b946df4bc924205e7d767f7f34f3e63b0a/packages/server/src/integrations/utils/utils.ts#L173

The datasource ID will always start with `datasource_` - if you look at the document type that is true for DS+ and DS, all external tables will always have that at the start - internal tables will always start with `ta_`.